lms/djangoapps/verify_student/views.py

```diff
@@ -493,6 +488,13 @@ def _redirect_if_necessary(
             else:
                 url = reverse('verify_student_start_flow', kwargs=course_kwargs)
 
+        if message in [self.FIRST_TIME_VERIFY_MSG, self.UPGRADE_MSG] and user.is_active:
+            # If the user is active and ecommerce service is enabled redirect the user to
+            # the ecommerce checkout page.
+            ecommerce_service = EcommerceService()
+            if ecommerce_service.is_enabled(user):
+                url = ecommerce_service.checkout_page_url(sku)
```

Comment:
If you move this to the calling function [above](https://github.com/edx/edx-platform/pull/12052/files#diff-6c54eb67d762f79cb6d3cec7307573dcR313) and redirect there, you can short circuit the rest of the logic that happens while trying to get to this point.